src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchListener.java

```diff
@@ -0,0 +1,130 @@
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature.twitch;
+
+import net.engio.mbassy.listener.Handler;
+import org.kitteh.irc.client.library.Client;
+import org.kitteh.irc.client.library.element.CapabilityState;
+import org.kitteh.irc.client.library.element.Channel;
+import org.kitteh.irc.client.library.element.MessageTag;
+import org.kitteh.irc.client.library.event.capabilities.CapabilitiesSupportedListEvent;
+import org.kitteh.irc.client.library.event.client.ClientReceiveCommandEvent;
+import org.kitteh.irc.client.library.exception.KittehServerMessageException;
+import org.kitteh.irc.client.library.feature.filter.CommandFilter;
+import org.kitteh.irc.client.library.feature.twitch.event.ClearChatEvent;
+import org.kitteh.irc.client.library.feature.twitch.event.GlobalUserStateEvent;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.BanDuration;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.BanReason;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.Color;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.DisplayName;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.EmoteSets;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.MsgId;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.Turbo;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.UserId;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.UserType;
+import org.kitteh.irc.client.library.util.Sanity;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Helpful things.
+ */
+public class TwitchListener {
+    /**
+     * Capability to receive commands.
+     */
+    public static final String CAPABILITY_COMMANDS = "twitch.tv/commands";
+
+    /**
+     * Capability to receive JOIN, MODE, NAMES, and PART.
+     */
+    public static final String CAPABILITY_MEMBERSHIP = "twitch.tv/membership";
+
+    /**
+     * Capability to receive tags.
+     */
+    public static final String CAPABILITY_TAGS = "twitch.tv/tags";
+
+    private final Client client;
+
+    public TwitchListener(@Nonnull Client client) {
+        this.client = Sanity.nullCheck(client, "Client cannot be null");
+        client.getEventManager().registerEventListener(this);
```

Comment:
I assume this class is meant to be used like: `client.getEventManager().registerEventListener(new TwitchListener(client));`, so this seems weird to do.